Automattic/jetpack
# My Jetpack: Add Product class. Add REST classes. refactoring...

On this PR:
* Add the `Products` class initially on this commit https://github.com/Automattic/jetpack/pull/22348/commits/c485736907344847cc4be3c06a29c283bc6b63ad, but also 
* Add a `REST_Products` class
* Refact the code adding a new `REST_Purchases` class.

Fixes #

#### Changes proposed in this Pull Request:

* Add Product, REST_Products and Rest_Purchases classes.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Apply these changes
* Products are initially exposed to the client via the `myJetpackInitialState` global var. Confirm the data is properly exposed there:
<img src="https://user-images.githubusercontent.com/77539/149980315-781875ea-b179-4d19-9b8b-6d108e08b0e6.png" width="600px" />

* Confirm the `[GET}` `/site/purchases` and `/site/products` work as expected. You can hardcode momentarily the permissions checking method returning True in order to check them basically using a client.

`testing-site/wp-json/my-jetpack/v1/site/products` | `testing-site/wp-json/my-jetpack/v1/site/purchases`
-----|-----
<img src="https://user-images.githubusercontent.com/77539/149980863-3b1980e8-02d8-4477-a7c6-927596102640.png"  /> | ![image](https://user-images.githubusercontent.com/77539/149981051-5bf07cfd-3222-4544-baeb-8c35ced4dd4f.png)
